devicetypes/smartthings/zigbee-ceiling-fan-light.src/led-fan-lightings.groovy

```diff
@@ -0,0 +1,245 @@
+				attributeState "turningOn", label: '${name}', action: "off", icon: "st.switches.light.on", backgroundColor: "#00A0DC", nextState: "turningOff"
+				attributeState "turningOff", label: '${name}', action: "on", icon: "st.switches.light.off", backgroundColor: "#ffffff", nextState: "turningOn"
+			}
+			tileAttribute("device.level", key: "SLIDER_CONTROL") {
+				attributeState "level", action: "switch level.setLevel"
+			}
+		}
+
+		standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", label: "", action: "refresh.refresh", icon: "st.secondary.refresh"
+		}
+		
+		main(["switch"])
+		details(["switch", "refresh", "switchLevel"])
+	}
+}
+
+private getFAN_CLUSTER_VALUE() { 0x0202 }
+private getFAN_STATUS_VALUE() { 0x0000 }
+private getON_OFF_CLUSTER_VALUE() { 0x0006 }
+
+def parse(String description) {
+	// Parse incoming device messages to generate events
+	log.debug "description is $description"
+	def event = zigbee.getEvent(description)
+	def zigbeeMap = zigbee.parseDescriptionAsMap(description)
+	if (event) {
+		sendEvent(event)
+	} else if (description?.startsWith('read attr -')) {
+		if (zigbeeMap.clusterInt == FAN_CLUSTER_VALUE &&
+		    zigbeeMap.attrInt == FAN_STATUS_VALUE) {		
+			log.debug "read attribute event for fan cluster attrib FAN_STATUS"
+			def childDevice = getChildDevices()?.find {
+				//find light child device
+				log.debug "parse() child device found"
+				it.device.deviceNetworkId == "${device.deviceNetworkId}-Fan" 
+			}          
+			event.displayed = true
+			zigbeeMap.name = "fanSpeed"
+			childDevice.sendEvent(zigbeeMap)
+			if (zigbeeMap.value == "00") {
+				log.debug "fan_off => switch off"
+				childDevice.sendEvent(name: "switch", value: "off")
+				childDevice.sendEvent(name: "level", value: 0)	// For cloud to cloud device UI update
+			} else {
+				log.debug "fan_on => switch on"
+				childDevice.sendEvent(name: "switch", value: "on")
+				def int_v  = zigbeeMap.value as Integer
+				int_v = int_v * 25
+				int_v = int_v > 100 ? 100 : int_v
+				log.debug "child device Level set $int_v"
+				childDevice.sendEvent(name: "level", value: int_v)	// For cloud to cloud device UI update
+			}
+		}		
+	} else {
+		def cluster = zigbee.parse(description)
+		if (cluster && 
+		    cluster.clusterId == ON_OFF_CLUSTER_VALUE &&
+		    cluster.command == 0x07) {
+			if (cluster.data[0] == 0x00) {
+				log.debug "ON/OFF REPORTING CONFIG RESPONSE: " + cluster
+				sendEvent(name: "checkInterval", value: 60 * 12, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+			} else {
+				log.warn "ON/OFF REPORTING CONFIG FAILED- error code:${cluster.data[0]}"
+			}
+		}
+	}
+}
+
+def off() {
+	zigbee.off()
+}
+
+def on() {
+	zigbee.on()
+}
+
+def setLevel(value, rate = null, device = null) {
+	if (device == null) {
+		zigbee.setLevel(value)
```

Comment:
Oh, sure. It is the most important part of this code. For separate operation between dimming and fan speed "device" added for it. Because of the separator "device",  "LEVEL_CONTROL_CLUSTER" can operates separately.